config/master/aws-k8s-cni-cn.yaml

```diff
@@ -154,6 +155,7 @@
           "name": "cni-bin-dir"
       "priorityClassName": "system-node-critical"
       "serviceAccountName": "aws-node"
+      "terminationGracePeriodSeconds": 10
```

Comment:
The default `terminationGracePeriodSeconds` is 30 seconds. The CNI does not really need any time to shut down, so reducing this to `10s`.

Reply:
why not zero (exit immediately)?

Reply:
Here I did set it at 10s on purpose to avoid leaking ENIs. There is a [shutdown hook]() and that flag is checked before [trying to free](https://github.com/aws/amazon-vpc-cni-k8s/blob/e38a4d8a8a9358f9b1e24f465a8d42a95a837533/pkg/ipamd/ipamd.go#L480-L483) or [add](https://github.com/aws/amazon-vpc-cni-k8s/blob/e38a4d8a8a9358f9b1e24f465a8d42a95a837533/pkg/ipamd/ipamd.go#L569-L572) an ENI. If the shutdown happens just when ipamd has detached an ENI, but before it has deleted it, we would "leak" it. It usually takes around 2 seconds for the detach to complete, but p99 is 10s according to EC2. Same for creating ENIs, we want the create to complete and the ENI to be attached to the node (with the delete on termination flag set) so that it gets cleaned up if the node terminates.

That said, i don't think we have done enough testing on this.

Reply:
(:rocket: This is exactly the sort of analysis I was hoping to capture)